### DISCUSSION TITLE
Inverter inconsistent responses/stops responding, error on start (soft bus A06) (Must 1kW inverter)

### DISCUSSION THREAD

**Discussion by sstepane** [vladyspavlov/esphome-must-inverter#10] [General]
Hi,

Have been observing for some time diy dongle with Must (MP1900, MP19-1012VM exact model) found out that:

- 5s poll interval is too often and causes either "duplicate command" or inconsistent responses, 30s is just fine.
Also at some point had to restart inverter as dongle stopped working. after resetting dongle inverter did not respond to polls whatsoever until restarted, so increasing poll time seems to be fixed this issue.

- if inverter restarts (especially it runs out of battery and restarts after mains are live) fast dongle boot up and polling might cause inverter soft bus error (A06). Setting on_boot delay for, say, 180 seconds fixes this behavior.

As a side note have been trying to put lambdas on some "list type" responses so the dongle will respond with meaningfull strings instead of 0-5 values - does not work well with 8266 type esp. either bug or "too much" processing for dongle mcu (8266). It turns out to be easier to be made by "template" type values in HA instead.

Drop a reply if "on_boot" or "template" points need to be clarified further.

p.s. your's inverter might not experience such behavior, as mine model is cheapest possible and could be made with way cheaper mcu, but given classic chinese mfg's behavior - it might be one mcu for all models.

**Comment by vladyspavlov** [comment]
> * 5s poll interval is too often and causes either "duplicate command" or inconsistent responses, 30s is just fine.

Have you tried using `register_count` to optimize register reads?
https://esphome.io/components/sensor/modbus_controller.html#configuration-variables

> does not work well with 8266 type esp

What about ESP32? Have you tried?

**Reply by sstepane** [reply]
> > * 5s poll interval is too often and causes either "duplicate command" or inconsistent responses, 30s is just fine.
> 
> Have you tried using `register_count` to optimize register reads? https://esphome.io/components/sensor/modbus_controller.html#configuration-variables
Nope. But if no one else experiencing this - reason seems obvious...
> 
> > does not work well with 8266 type esp
> 
> What about ESP32? Have you tried?
No. Not critical, but pretty sure it's a software optimization within esp. Would not believe that "switch" type logic within C/C++ could cause slow performance on pretty fast mcu...
